packages/amplify-graphql-transformer-core/src/transformer-context/resolver.ts

```diff
@@ -162,14 +162,59 @@ export class TransformerResolver implements TransformerResolverProvider {
       slotEntry = [];
     }
 
-    slotEntry.push({
-      requestMappingTemplate,
-      responseMappingTemplate,
-      dataSource,
-    });
+    if (!this.updateSlotIfExists(slotName, requestMappingTemplate, responseMappingTemplate)) {
+      slotEntry.push({
+        requestMappingTemplate,
+        responseMappingTemplate,
+        dataSource,
+      });
+    }
     this.slotMap.set(slotName, slotEntry);
   };
 
+  updateSlotIfExists = (
+    slotName: string,
+    requestMappingTemplate?: MappingTemplateProvider,
+    responseMappingTemplate?: MappingTemplateProvider,
+  ): boolean => {
+    const slotEntires = this.slotMap.get(slotName)!;
+    let slotIndex = 1;
+
+    if (!slotEntires) {
+      return false;
+    }
+
+    for (const slotItem of slotEntires) {
+      const [slotItemRequestMappingTemplate, slotItemResponseMappingTemplate] = [
+        (slotItem.requestMappingTemplate as any)?.name ?? '',
+        (slotItem.responseMappingTemplate as any)?.name ?? '',
+      ]
+        .map(name => name.replace('{slotName}', slotName).replace('{slotIndex}', slotIndex));
+
+      // If both request and response mapping templates are inline, return false
+      if (slotItemRequestMappingTemplate === '' && slotItemResponseMappingTemplate === '') {
+        return false;
+      }
+
+      // If name matches, then it is an overridden resolver
+      if (
+        slotItemRequestMappingTemplate === ((requestMappingTemplate as any)?.name ?? '')
```

Comment:
In the case the user only wants to override the request mapping template but not the response VTL, will this logic work?

Reply:
Yes. For example, if response VTL is not overridden, `(responseMappingTemplate as any)?.name` would be `undefined`. In that case, the slot's response VTL is preserved.

```js
slot.responseMappingTemplate = (responseMappingTemplate as any)?.name
        ? responseMappingTemplate
        : slot.responseMappingTemplate;
```